experimental/devices/pca9548/pca9548.go

```diff
@@ -0,0 +1,155 @@
+// Copyright 2018 The Periph Authors. All rights reserved.
+// Use of this source code is governed under the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+package pca9548
+
+import (
+	"errors"
+	"strconv"
+	"sync"
+
+	"periph.io/x/periph/conn/i2c"
+	"periph.io/x/periph/conn/i2c/i2creg"
+	"periph.io/x/periph/conn/physic"
+)
+
+// Dev is handle to a pca9548 I²C Multiplexer.
+type Dev struct {
+	c       i2c.Bus
+	address uint16
+	names   []string
+	ports   uint8
+	// mu guards port
+	mu   sync.Mutex
+	port uint8
+}
+
+// DefaultOpts is the recommended default options.
+var DefaultOpts = Opts{Address: 0x70, Ports: 8}
+
+// Opts is the pca9548 configuration.
+type Opts struct {
+	// Address pca9548 I²C Address.
+	Address uint16
+	// Ports number of physical ports on I²C Multiplexer.
+	Ports uint8
+}
+
+// Register creates a new handel to a pca9548 I²C multiplexer, and registers
+// port names with the host. These ports can then be used as any other i2c.Bus.
+// The registered port names are in the form: i2cmux/mux-ADD-I where ADD is the
+// multiplexer I²C address in hex and I is the port number.
+// example: "i2cmux-70-0" and "mux-70-0".
+func Register(bus i2c.Bus, opts *Opts) (*Dev, error) {
+	d := &Dev{
+		c:       bus,
+		port:    0xFF,
+		address: opts.Address,
+		ports:   opts.Ports,
+	}
+
+	for i := uint8(0); i < opts.Ports; i++ {
+		portID := strconv.FormatUint(uint64(i), 10)
+		addrStr := strconv.FormatUint(uint64(opts.Address), 16)
+		name := addrStr + "-" + portID
+		opener := newOpener(d, i)
+		if err := i2creg.Register("i2cmux-"+name, []string{"mux-" + name}, int((opts.Address*10)+uint16(i)), opener); err != nil {
```

Comment:
Just for the behavior described about. I like that they are ready to use. But you make me think that I should check to see the IC is there before registering the ports. That way it can fail right there rather than later in the middle of unrelated code.